### PR TITLE
test: latency of internal payments within fed

### LINKED ISSUE #3695: Add latency test for internal swap
[latency tests](https://github.com/fedimint/fedimint/blob/4a6a580f4eeccc7f95baa8c693c2f5ae52900a24/devimint/src/main.rs#L27) should have coverage for internal swap between two clients in the same federation

### PR DESCRIPTION
**Pull request by okjodom**
Adds latency test coverage to transactions between two fedimint clients in a federation.
From initial tests, turns out these internal transactions are currently much slower than LN sends / receives!
A follow up here is to investigate and improve internal payments perf

```rs
================= RESULTS ==================
REISSUE: min: 1.7s, avg: 1.8s, median: 1.8s, p90: 1.9s, max: 2.0s, sum: 54.1s
LN SEND: min: 2.9s, avg: 3.1s, median: 3.1s, p90: 3.2s, max: 4.4s, sum: 62.8s
LN RECV: min: 1.9s, avg: 2.2s, median: 2.2s, p90: 3.0s, max: 3.1s, sum: 44.1s

FM PAY: min: 4.2s, avg: 4.7s, median: 4.4s, p90: 5.0s, max: 9.4s, sum: 93.5s
```

closes #3695

### CODE REVIEW

File: devimint/src/federation.rs [hunk 1]
```diff
@@ -242,6 +242,23 @@ impl Federation {
...
+        self.bitcoind
+            .send_to(deposit_address.to_owned(), amount)
+            .await?;
+        self.bitcoind.mine_blocks(21).await?;
+
+        cmd!(client, "await-deposit", deposit_operation_id)
+            .run()
+            .await?;
+        Ok(())
+    }
```
Comment (elsirion) [inline devimint/src/federation.rs:260]: The `pegin` function should be deduplicate with this one.

Comment (okjodom) [reply devimint/src/federation.rs:260]: Made `pegin` delegate to `pegin_client`. I should fully remove pegin in a rewrite of #3698 which is now quite stale

**Review by elsirion — APPROVED** [review]
Looks good :)

**Comment by justinmoon** [comment]
Nice. @okjodom could you make an issue for the investigation?

**Review by justinmoon — APPROVED** [review]

**Pull request merged**